src/gt4sd/frameworks/enzeptional/processing.py

```diff
@@ -210,27 +290,83 @@ def mutate_sequence_with_variant(sequence: str, variant: str) -> str:
 
 def sanitize_intervals(intervals: List[Tuple[int, int]]) -> List[Tuple[int, int]]:
     """Sanitize intervals merging overlapping ones and sorting them.
-
     Args:
         intervals: intervals to sanitize.
-
     Returns:
         sorted and non overlapping intervals.
     """
     sorted_intervals = sorted(intervals, key=lambda interval: interval[0])
-    merged_intervals = [sorted_intervals[0]]
-    for current in sorted_intervals:
-        previous_end = merged_intervals[-1][1]
+    merged_intervals = []
+    previous = sorted_intervals[0]
+    for pos in range(1, len(sorted_intervals)):
+        current = sorted_intervals[pos]
+        previous_end = previous[1]
         if current[0] <= previous_end:
             previous_end = max(previous_end, current[1])
+            previous = (previous[0], previous_end)
+            if pos == len(sorted_intervals) - 1:
+                merged_intervals.append(previous)
         else:
+            merged_intervals.append(previous)
             merged_intervals.append(current)
     return merged_intervals
 
 
+def round_up(number):
+    return int(number) + (number % 1 > 0)
+
+
+def sanitize_intervals_with_padding(
+    intervals: List[Tuple[int, int]], sequence_length: int = 64
+) -> List[Tuple[int, int]]:
+    """Sanitize intervals padding the intervals that contain less then 8 AAs to 8.
+
+    Args:
+        intervals: intervals to sanitize.
+
+    Returns:
+        sorted, non overlapping, and padded intervals.
+    """
+
+    clean_intervals = sanitize_intervals(intervals)
+    padded_intervals = []
+    for start, end in clean_intervals:
```

Comment:
parametrize the 8, add a default argument `minimum_interval_length: int = 8` and use it in the function